src/drivers/amqp_interop/Queue.php

```diff
@@ -223,6 +223,28 @@ public function init()
         Event::on(BaseApp::class, BaseApp::EVENT_AFTER_REQUEST, function () {
             $this->close();
         });
+
+        if (extension_loaded('pcntl') && PHP_MAJOR_VERSION >= 7) {
+            // https://github.com/php-amqplib/php-amqplib#unix-signals
+            $signals = [SIGTERM, SIGQUIT, SIGINT, SIGHUP];
+            
+            foreach($signals as $signal) {
+                $oldHandler = null;
+                // This got added in php 7.1 and might not exist on all supported versions
+                if(function_exists('pcntl_signal_get_handler')) {
+                    $oldHandler = pcntl_signal_get_handler($signal);
+                }
+
+                pcntl_signal($signal, function ($signal) use($oldHandler) {
```

Comment:
1. the callback could be declared as `static` function
1. `use($oldHandler)` could use a space `use ($oldHandler)`